doc/man3/OSSL_HTTP_REQ_CTX.pod

```diff
@@ -0,0 +1,172 @@
+gets populated with the B<BIO> to send the request to (I<wbio>), the B<BIO> to
+read the response from (I<rbio>), the request method (I<method_GET>, which may
+be 1 to indicate that the C<GET> method is to be used, or 0 to indicate that
+the C<POST> method is to be used), the maximum expected response length
+(I<max_resp_len>), a response timeout measure in seconds (I<timeout>), the
+expected MIME content type of the response (I<expected_content_type>), and a
+flag indicating that the response is expected to be a DER encoded ASN.1
+structure (I<expect_asn1>).
+The allocated context structure is also populated with an internal allocated
+memory B<BIO>, which collects the HTTP request and additional headers as text.
+The returned context should only be used for a single HTTP request/response.
+
+OSSL_HTTP_REQ_CTX_free() frees up the HTTP request context I<rctx>.
+The I<wbio> and I<rbio> are not free'd and it is up to the application
+to do so.
+
+OSSL_HTTP_REQ_CTX_header() adds an HTTP request line to the request context.
+The request command itself becomes C<GET> or C<POST> depending on the value
+of I<method_GET> in the OSSL_HTTP_REQ_CTX_new() call.  I<server> and I<port>
+may be set to indicate a proxy server and port that the request should go
+through, otherwise they should be left NULL.  I<path> is the HTTP request path;
+if left NULL, C</> is used.
+
+OSSL_HTTP_REQ_CTX_add1_header() adds header I<name> with value I<value> to the
+context I<rctx>. It can be called more than once to add multiple headers.
+For example, to add a C<Host> header for C<example.com> you would call:
+
+ OSSL_HTTP_REQ_CTX_add1_header(ctx, "Host", "example.com");
+
+OSSL_HTTP_REQ_CTX_i2d() finalizes the HTTP request context by adding the DER
+encoding of I<req>, using the ASN.1 template I<it> to do the encoding.  The
+HTTP header C<Content-Length> is automatically filled out, and the HTTP header
+C<Content-Type> is also added, with I<content_type> as its value.  All of this
+ends up in the internal memory B<BIO>.  This requires that the request type be
+C<POST>, i.e. that I<method_GET> is 0 in the OSSL_HTTP_REQ_CTX_new() call.
+
+OSSL_HTTP_REQ_CTX_nbio() performs the exchange of request and response via HTTP,
+using the I<rbio> and I<wbio> that were given in the OSSL_HTTP_REQ_CTX_new()
+call.  When successful, the contents of the internal memory B<BIO> is replaced
+with the contents of the HTTP response, without the response headers.
+
+OSSL_HTTP_REQ_CTX_sendreq_d2i() calls OSSL_HTTP_REQ_CTX_nbio(), and DER decodes
+the received response using the ASN.1 template I<it>.
+
+OSSL_HTTP_REQ_CTX_set_max_response_length() sets the maximum response length
+for I<rctx> to I<len>. If the response exceeds this length an error occurs.
+If not set a default value of 100k is used.
+
+OSSL_HTTP_REQ_CTX_get0_mem_bio() returns the internal memory B<BIO>.  This can
+be used to affect the HTTP request text.  I<Use with caution!>
```

Comment:
BTW, as I learned, in POD files better place each sentence on a separate line.

Reply:
This was disputed against at some point, and to that, I shrugged with the response thaat as we don't have the endless dispute about number of spaces after a period, I'm OK with either.

Reply:
OK by me, though that having each sentence on a separate line makes the file at least better maintainable (less brittle on small changes, and thus more easy to review those).

Reply:
Ah well, I consider this a discussion and change for another time